File: Algorithm/baekjoon/swift/BJ1080.swift

//1080 런타임 에러..
func compare(input: [[Bool]], output: [[Bool]]) -> Bool{
    for i in 0..<input.count{
        for j in 0..<input[0].count{
            if(input[i][j] != output[i][j]){
                return false
            }
        }
    }
    return true
}
func flip( input: inout [[Bool]], i: Int, j: Int){
    for r in i..<i+3{
        for c in j..<j+3{
            if r < input.count && c < input[0].count{
                input[r][c] = !input[r][c]
            }
        }
    }
}

var input = readLine()
if let input = input{
    let values = input.split(separator: " ");
    let row = Int(values[0])!
    let col = Int(values[1])!
    
    var input : [[Bool]] = []
    for _ in 0..<row{
        let line = readLine()!.map({(c: Character) -> Bool in
            if(c == "0"){ return false}
            else{ return true}
        })
        
        input.append(line)
    }
    
    var output : [[Bool]] = []
    for _ in 0..<row{
        let line = readLine()!.map({(c: Character) -> Bool in
            if(c == "1"){ return true }
            else{ return false}
        })
        output.append(line)
    }
    
    var result = 0
    for i in 0..<row-2{
        for j in 0..<col-2{
            if input[i][j] != output[i][j]{
                result += 1
                flip(input: &input, i: i, j: j)
            }
        }
    }
    
    if(compare(input: input, output: output)){
        print(result)
    }
    else{
        print(-1)
    }
}

File: Algorithm/baekjoon/java/BJ1080.java
package baekjoon.java;

/**
 * Created by changmin on 2018. 1. 12..
 */
import java.util.*;
public class BJ1080 {
   static int change(boolean[][] input, boolean[][] output){
      int count = 0;

      for(int i = 0; i < input.length - 2; ++i){
         for(int j = 0; j < input[0].length - 2; ++j){
            if(input[i][j] != output[i][j]){
               flip(input, i , j);
               count += 1;
            }
         }
      }

      for(int i = 0 ; i < input.length; ++i){
         for(int j = 0; j < input[0].length; ++j){
            if(input[i][j] != output[i][j]){
               return -1;
            }
         }
      }

      return count;
   }

   static void flip(boolean[][] input, int row, int col){
      for(int i = row; i < row + 3; ++i){
         for(int j = col; j < col + 3; ++j){
            if(i < input.length && j < input[0].length){
               input[i][j] = !input[i][j];
            }
         }
      }
   }

   public static void main(String[] args){
      Scanner sc = new Scanner(System.in);
      int row = sc.nextInt();
      int col = sc.nextInt();
      sc.nextLine();

      boolean input[][] = new boolean[row][col];
      boolean output[][] = new boolean[row][col];

      for(int i = 0; i < row; ++i){
         String in = sc.nextLine();
         for(int j = 0; j < col; ++j){
             if(in.charAt(j) == '1'){
                input[i][j] = true;
             }
         }
      }

      for(int i = 0; i < row; ++i){
         String in  = sc.nextLine();
         for(int j = 0; j < col; ++j){
             if(in.charAt(j) == '1') {
                output[i][j] = true;
             }
         }
      }

      int result = change(input, output);
      System.out.println(result);
   }
}
